fearless/Common/Services/ExtrinsicService/ExtrinsicService.swift

```diff
@@ -122,38 +215,60 @@ extension ExtrinsicOperationFactory: ExtrinsicOperationFactoryProtocol {
             dependingOn: nonceOperation,
             codingFactoryOperation: codingFactoryOperation,
             customClosure: closure,
+            numberOfExtrinsics: numberOfExtrinsics,
             signingClosure: signingClosure
         )
 
         builderOperation.addDependency(nonceOperation)
         builderOperation.addDependency(codingFactoryOperation)
 
-        let infoOperation = JSONRPCListOperation<RuntimeDispatchInfo>(
-            engine: engine,
-            method: RPCMethod.paymentInfo
-        )
+        let feeOperationList: [JSONRPCListOperation<RuntimeDispatchInfo>] =
+            (0 ..< numberOfExtrinsics).map { index in
+                let infoOperation = JSONRPCListOperation<RuntimeDispatchInfo>(
+                    engine: engine,
+                    method: RPCMethod.paymentInfo
+                )
+
+                infoOperation.configurationBlock = {
+                    do {
+                        let extrinsic = try builderOperation.extractNoCancellableResultData()[index]
+                            .toHex(includePrefix: true)
+                        infoOperation.parameters = [extrinsic]
+                    } catch {
+                        infoOperation.result = .failure(error)
+                    }
+                }
+
+                infoOperation.addDependency(builderOperation)
+
+                return infoOperation
+            }
 
-        infoOperation.configurationBlock = {
-            do {
-                let extrinsic = try builderOperation.extractNoCancellableResultData().toHex(includePrefix: true)
-                infoOperation.parameters = [extrinsic]
-            } catch {
-                infoOperation.result = .failure(error)
+        let wrapperOperation = ClosureOperation<[FeeExtrinsicResult]> {
+            feeOperationList.map { feeOperation in
+                if let result = feeOperation.result {
+                    return result
+                } else {
+                    return .failure(BaseOperationError.parentOperationCancelled)
+                }
             }
         }
 
-        infoOperation.addDependency(builderOperation)
+        feeOperationList.forEach { feeOperation in
+            wrapperOperation.addDependency(feeOperation)
+        }
 
         return CompoundOperationWrapper(
-            targetOperation: infoOperation,
-            dependencies: [nonceOperation, codingFactoryOperation, builderOperation]
+            targetOperation: wrapperOperation,
+            dependencies: [nonceOperation, codingFactoryOperation, builderOperation] + wrapperOperation.dependencies
```

Comment:
Please, put manually `feeOperationList` otherwise we can at some point add an operation twice